### ISSUE TITLE
Multi-GPU training for detector

### ISSUE THREAD

**Issue by saswat0** [aosokin/os2d#45]
Can we train the CNN detector using multiple GPUs? The available arguments cause the program to hang after loading the data and model. I'm using the following line to initiate the process.
```
python -m torch.distributed.launch --nproc_per_node=2  --use_env train_detector.py  --config-file config/e2e_faster_rcnn_R_50_FPN_1x_multiscale_noClasses.yaml DATASETS.TRAIN [\"grozi-train\"] DATASETS.TEST [\"grozi-val-all\"] INPUT.MIN_SIZE_TRAIN [480,600,768,960,1152,1344,1536] INPUT.MAX_SIZE_TRAIN 2048 INPUT.MIN_SIZE_TEST 960 INPUT.MAX_SIZE_TEST 1280 OUTPUT_DIR output/exp0000-R-50-noCl-grozi
```

**Comment by aosokin** [comment]
Hi,this probably can done but will likely require adjusting the code. We've never tried training on multiple GPUs.

**Comment by saswat0** [comment]
Okay. In my case, I have a 48GB GPU, but the training occupies only 8GB, thereby taking three days to train the detector. How were you able to do this in a shorter time?

**Comment by aosokin** [comment]
But what is the processing load of your GPU? if it is low you can try increasing batch size.
Another idea: it might be fine to train for significantly fewer iterations, you just need to monitor the behaviour of the validation loss to stop the process early.

**Comment by saswat0** [comment]
GPU runs at 100% capacity, but most memory is left idle. Increasing the batch size in the config file isn't reflected in the final parameters. Did you face this issue while experimenting?
For the second approach, the current code doesn't have a tfboard support. Should I monitor the logs instead?

**Comment by aosokin** [comment]
> GPU runs at 100% capacity, but most memory is left idle. Increasing the batch size in the config file isn't reflected in the final parameters. Did you face this issue while experimenting?

It sounds weird, changing `train.batch_size` and `train.class_batch_size` should  definitely change the training process - at least the GPU memory usage should go up.
However, if the GPU is already at 100% simply changing batch size is not likely to increase training speed.

> For the second approach, the current code doesn't have a tfboard support. Should I monitor the logs instead?

Yes, the code does not have tensorboard but it includes another visualization tool: [os2d/utils/plot_visdom.py](https://github.com/aosokin/os2d/blob/master/os2d/utils/plot_visdom.py)

**Comment by saswat0** [comment]
Got it, thanks!
